src/renderer/qssWidget/js/qssQuickFoldersWidget.js

```diff
@@ -0,0 +1,173 @@
+ * https://github.com/GPII/universal/blob/master/LICENSE.txt
+ */
+
+/* global fluid */
+
+"use strict";
+(function (fluid) {
+    var gpii = fluid.registerNamespace("gpii");
+
+    /**
+     * QSS Quick folders widget
+     */
+    fluid.defaults("gpii.qssWidget.quickFolders", {
+        gradeNames: ["fluid.viewComponent", "gpii.psp.selectorsTextRenderer"],
+
+        selectors: {
+            folderSearch: ".flc-qssQuickFoldersWidget-folderSearch",
+            settingTitle: ".flc-qssSearchWidget-settingTitle",
+            searchField: ".flc-search",
+            searchButton: ".flc-searchButton",
+            labelTitle: ".flc-qssSearchWidget-label",
+            errorMessage: ".flc-qssSearchWidget-errorMessage"
+        },
+
+        enableRichText: true,
+
+        model: {
+            setting: {},
+            value: "{that}.model.setting.value",
+            messages: {
+                labelTitle: "{quickFolders}.model.messages.label",
+                searchButtonLabel: "{quickFolders}.model.messages.searchButtonLabel",
+                alertLabel: "{quickFolders}.model.messages.alertLabel"
+            },
+            errorMessageSelector: "{that}.options.selectors.errorMessage",
+        },
+
+        listeners: {
+            onCreate: {
+                func: "{that}.onCreate"
+            }
+        },
+        invokers: {
+            onCreate: {
+                funcName: "gpii.qssWidget.quickFolders.onCreate",
+                args: ["{gpii.qssWidget.quickFolders}.errorMessage"] // sending the errorMessage component
+            }
+        },
+
+        components: {
+            searchField: {
+                type: "gpii.psp.widgets.textfield",
+                container: "{that}.dom.searchField",
+                options: {
+                    model: {
+                        value: "{gpii.qssWidget.quickFolders}.model.value"
+                    },
+                    invokers: {
+                        onSearch: {
+                            funcName: "gpii.qssWidget.quickFolders.onSearch",
+                            args: [
+                                "{that}.model.value", // search field value
+                                "{gpii.qssWidget.quickFolders}.errorMessage", //errorMessage component
+                                "{gpii.qssWidget.quickFolders}.options.siteConfig.morphicQuickFolderPath" // siteConfig's base folder
+                            ]
+                        }
+                    }
+                }
+            },
+            searchButton: {
+                type: "gpii.psp.widgets.button",
+                container: "{that}.dom.searchButton",
+                options: {
+                    model: {
+                        label: "{gpii.qssWidget.quickFolders}.model.messages.searchButtonLabel"
+                    },
+                    invokers: {
+                        onClick: "{gpii.qssWidget.quickFolders}.searchField.onSearch"
```

Comment:
Again onClick should be an event

Reply:
The onClick invoker seems to be required in the gpii.psp.widgets.button itself, as you can see in - widgets.js:320.

Reply:
In that case we should convert the button's invoker to an event too - invoker named "onClick" with an implementation of fluid.identity is a classic anti-pattern : P
We then convert the button's "onCreate.addClickHandler" to invoke "{that}.onClick.fire"

Reply:
@amb26 I totally agree that its a broken pattern, that's for sure! And I've tried your method, which seems great - it seems like the button will continue working as is, and we can use it as a event in the current widget. But its not like that, I can make it work somehow in the current widget, but all other buttons are stopping working. Of course that can be fixed, by reworking all of the broken buttons, but I'm not sure if this cannot be moved into another task and just close and merge this PR as is? We can discuss it in today's meeting, I will put it into the agenda.

Reply:
Written up as https://issues.gpii.net/browse/GPII-3776 for the next cycle